### DISCUSSION TITLE
Release v0.19.0 Beta

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#3898] [Announcements]
# General Update

I know it's been a long time since a formal update. I'll try to explain my thoughts regarding this in a later blog post. Despite the silence, there's a wealth of changes in this release and I hope this will make up for the wait.

# Local API

Our local API has undergone a massive overhaul. We have switched all local API hooks to use [YouTube.js](https://github.com/LuanRT/YouTube.js) instead of our separate modules for each function. This should hopefully allow us to fix any issues with the local API faster than what we could beforehand. Any relevant module that we have created has been merged into YouTube.js for a more uniformed experience. A massive thank you to @LuanRT for their help and support on YouTube.js.

This is a very large change to how we handle things. If you happen to see anything that seems broken compared to the last release, please consider creating a ticket over on our GitHub so that we can look at it.

## Features
* Mobile devices can now automatically enter fullscreen when rotated to landscape. Find the toggle in the player settings. By @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/2808
* You can now share channel links using the new share button within a channel page by @Alhuin in https://github.com/FreeTubeApp/FreeTube/pull/2859
* You can now right click on a video to open it in a new window or copy YouTube / Invidious links via the context menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2873 and https://github.com/FreeTubeApp/FreeTube/pull/2957
* You can now hide upcoming premieres in the distraction free settings by @miangraham in https://github.com/FreeTubeApp/FreeTube/pull/2853
* Added `Cmd / Ctrl + H` shortcut to go to history by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2870
* You can now hide the FreeTube logo within the theme settings by @theforbiddenpool in https://github.com/FreeTubeApp/FreeTube/pull/2893
* You can now jump between video chapters using the `Cmd or Ctrl + Left Arrow` and `Cmd or Ctrl + Right Arrow` keyboard shortcuts by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2912
* Playlists now support proper pagination (Local API only) by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2956
* Settings can now be password protected by @elshimone in https://github.com/FreeTubeApp/FreeTube/pull/2932
* You can now hide channels from various lists within the app by @petaded in https://github.com/FreeTubeApp/FreeTube/pull/2849
* You can now scrub through a video via scrolling. Find the toggle within settings. By @ai25 in https://github.com/FreeTubeApp/FreeTube/pull/2418
* Added support for [DeArrow](https://dearrow.ajay.app/), a service created by @ajayyy. DeArrow swaps the titles of a video with a more descriptive title. By @elshimone in https://github.com/FreeTubeApp/FreeTube/pull/2987 and by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3688
* You can now use AV1 video formats when available. Enable the toggle within player settings. By @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3105
* Viewing a channel now shows their Community Page by @GilgusMaximus in https://github.com/FreeTubeApp/FreeTube/pull/1568
* Viewing a channel now shows their Live Videos Page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3273
* Viewing a channel now shows their Shorts Videos Page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3533
* Viewing a channel now shows their Releases and Podcasts Pages by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3778
* You can now have comments automatically load when watching a video. Find the option within settings. By @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3352
* Added support to view hashtag pages by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3483 and https://github.com/FreeTubeApp/FreeTube/pull/3571
* The video player now supports multi-audio videos (Local API only) by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3563
* The Subscriptions page now has separate tabs to see shorts and live streams by @PrestonN in https://github.com/FreeTubeApp/FreeTube/pull/3725
* Added basic support for the PotPlayer, MPC-HC, MPC-BE external players by @trostboot in https://github.com/FreeTubeApp/FreeTube/pull/3798
* The search bar can now parse Invidious search URLs by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3849

## Fixes
* Fixed an issue with emoji's displaying within the profile selector by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2815
* Fixed an issue where chapter markers were slightly offset by their actual times by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2874
* Fixed an issue where cached subscription videos were not cleared when you remove all subscriptions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2905
* Fixed an issue where the Unsubscribe button would show in a channel page if parental controls were enabled by @elshimone in https://github.com/FreeTubeApp/FreeTube/pull/2906
* Fixed an issue where copying a URL link would sometimes provide an internal URL by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2951
* Fixed in issue where the import subscriptions link wouldn't open in your default browser by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2981
* Fixed an issue where some playlist links from YouTube were not being parsed properly by @petaded in https://github.com/FreeTubeApp/FreeTube/pull/2972
* Fixed an issue where `Cmd or Ctrl + C` wouldn't copy text when viewing a video by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3027
* Updated the Mastodon URL to the new account in the About view by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/3038
* Fixed an issue where searching within channels wouldn't work by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3062 and https://github.com/FreeTubeApp/FreeTube/pull/3073
* Fixed an issue where clicking on the currently selected trending tab would try to grab information again by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3070
* Fixed an issue where subtitles would sometimes not display within the video player by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3100
* Fixed an issue where `Shift + Enter` wouldn't open the search within a new window by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3113
* Fixed an issue where QT apps using the FreeTube protocol wouldn't work properly by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3119
* Fixed some styling issues in the playlist page by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/3123
* Fixed in issue where the channels page wouldn't populate with videos when sorting by popular and using the Invidious API by @camcam1773 in https://github.com/FreeTubeApp/FreeTube/pull/3126
* Fix an issue where the app would sometimes load on a non-existant second monitor by @kawanakaiku in https://github.com/FreeTubeApp/FreeTube/pull/3008
* Fixed an issue where external video players would sometimes not open within the Subscriptions page by @ArturWagnerBusiness in https://github.com/FreeTubeApp/FreeTube/pull/3172
* Fixed an issue where the Trending and Watch pages wouldn't allow certain arrow key events by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3170
* Fixed an issue where FreeTube couldn't parse channel URLs if they included tab information by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3241
* Fixed an issue where Live and Upcoming videos wouldn't show up within playlists by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3250
* Fixed an issue where Premieres wouldn't display properly in the Subscription page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3240 and https://github.com/FreeTubeApp/FreeTube/pull/3358
* Fixed an issue where session volume would not apply when opening a video in a new window by @predystopic-dev in https://github.com/FreeTubeApp/FreeTube/pull/3203
* Fixed an issue where FreeTube would permanently prevent the PCs screen saver from applying by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3286 and by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3557
* Fixed an issue with playing videos within playlists while using the Invidious API by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3294
* Fixed an issue where the chapter index wouldn't reset when changing videos by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3400
* Fixed an issue where watching a video would use your saved watch progress if progress saving is disabled by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3453
* Added better external link parsing for comments within the Local API by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3448
* Fixed an issue where the countdown to watch the next video in a playlist would still show up when you're at the end of a playlist and looping is disabled by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3362
* Fixed an issue with sorting upcoming subscriptions when using the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3485
* Fixed an issue with fetching more comment replies when using the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3489
* Fixed an issue where FreeTube wouldn't stay running in the background on MacOS when all windows were closed by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3526
* Fixed an issue with fetching more search results when using the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3486
* Fixed an issue where the user's name wouldn't show up or when a URL is missing from a badge within the live chat by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3558
* Fixed an issue where the cursor would move while navigating through search suggestions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3566
* Fixed an issue where the playlist thumbnail would not get proxied when using the Invidious API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3581
* Fixed an issue where the captions toggle keyboard shortcut wouldn't work for Legacy or Audio formats by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3580
* Fixed an issue where live videos within lists wouldn't have a save to playlist button by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3598
* Fixed an issue where custom emojis wouldn't display properly when using the Invidious API by @lamemakes in https://github.com/FreeTubeApp/FreeTube/pull/3658
* Fixed an issue where the Hide Channel Subscribers Distraction Free Setting would sometimes not work when using a secondary window by @Rikthepixel in https://github.com/FreeTubeApp/FreeTube/pull/3692
* Fixed an issue where several pages would continuously refresh when holding down certain keyboard shortcuts by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3709
* Fixed an issue where links within comments wouldn't be correct when using the Invidious API by @lamemakes in https://github.com/FreeTubeApp/FreeTube/pull/3711
* Fixed an issue where Hide Channels and Hide Premiere settings wouldn't work properly by @petaded in https://github.com/FreeTubeApp/FreeTube/pull/3673
* Fixed an issue where the Current Invidious Instance input field would remove characters while typing by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3776
* Fixed an issue where the watch page would break if the subscriber count is missing by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3794
* Fixed an issue with trying to send playlist information to external players with no index support by @trostboot in https://github.com/FreeTubeApp/FreeTube/pull/3797
* Fixed an issue when trying to import subscriptions that have terminated channels by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3816
* Fixed an issue with the focus box now showing up when using the tab key within the channel page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3850
* Fixed an issue where the search bar wasn't grabbing words from the suggestion list by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3793

## Changes
* Playlists now show video numbers by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2811
* The subscriber count on the watch page is now translated properly by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2841
* Several more elements in the app can be interacted with via middle click or right click by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2911 and https://github.com/FreeTubeApp/FreeTube/pull/2968
* Opening and Copying videos will include playlist information when available by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2848
* The "Load More Comments" button will only show up if there are more comments to grab by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2917
* The app menu now has history forward / back buttons by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2933
* The text input element has been given some accessibility improvements by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2709
* The profile selector now supports longer names by @emmayjiang in https://github.com/FreeTubeApp/FreeTube/pull/2964
* The subscribed channels page now better supports longer names by @elshimone in https://github.com/FreeTubeApp/FreeTube/pull/2988
* The Watch page has been given some accessibility improvements by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2986
* The Channel View has been given some accessibility improvementsby @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2984
* The app menu now has navigation items for the different side navigation items by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2965
* Channel handles are now parsed and shown within video descriptions instead of showing a raw URL by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3011
* Video history now remembers relevant playlist information when applicable by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3006
* The share menu in the playlists page has been updated by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3121
* You can now search by `Movie` type within the search bar by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3141
* Shorter YouTube Shorts videos should now display storyboars by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3128
* Manually scrubbing to sponsor spots in a video when Sponsorblock is turned out will now allow you to watch that sponsor by @lodwhet in https://github.com/FreeTubeApp/FreeTube/pull/3116
* All share buttons now use the secondary colour theme by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3169
* Grabbing subscriptions via RSS now display relevant time instead of absolute by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3216
* Exported files now use your local time instead of UTC in the file name by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3236
* A users member icon now shows up in the comments when using the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3293
* The Most Popular Tab in the side menu will now be hidden if the Invidious API is not selected and if backend fallback is disabled by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3339
* Improved default quality and quality selection logic by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3278
* A message will be shown when the Live Chat is unavailable by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3327
* You can now import YouTube Video History via the JSON export by @arlm in https://github.com/FreeTubeApp/FreeTube/pull/2958
* The frame by frame keyboard shortcut now supports live videos and 50fps videos by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3420
* Comments now show channel tags instead of the channel URL by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3351
* The playlist widget within the watch page will now auto scroll to the current video by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3399
* The chapters widget within the watch page will now auto scroll to the current chapter by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3610
* The distraction free settings are now sorted by subsections by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3591
* Playlists now use the provided playlist thumbnail instead of picking one ourselves (Local API Only) by @jadenet in https://github.com/FreeTubeApp/FreeTube/pull/3710
* Playlists now show the view count as well as the published date by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3767
* Subscriptions will now show the video duration when RSS is enabled if the information is available in your watch history by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/3839
* The Most Popular page will now show an error message if the instance you're using does not support it by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3841
* Added Improved Accessibility to the Profile Selector by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3863
* Added semantic roles to several elements within the app to make the site more accessible by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/3887
* Subscriptions are now cached on a per channel basis instead of a per profile basis by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3668
* Secondary windows are now updated quicker when watch history changes by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/3353

## Known Issues
* Proxying videos via Invidious is not working at the moment. See https://github.com/FreeTubeApp/FreeTube/issues/3892 and https://github.com/iv-org/invidious/issues/4045 for more information.

## Dropping Support for Older Windows Builds

A while back, Electron, which FreeTube is based on, dropped support for Windows 7, 8, and 8.1. Because of this, we are forced to do so as well. This release is the last release that will be supporting those operating systems and future versions of FreeTube will not work on those systems either. If you are still using any of these operating systems still, it is highly encouraged that you switch to a different operating system if at all possible.

## Looking for Translators!

We are actively looking for new translators! If you happen to know multiple languages and would like to see them added into FreeTube, we would love to have your contributions. If you know a language that FreeTube already supports, you can also contribute by making corrections or by adding missing translations. Regardless of where you stand, feel free to check out our [Weblate](https://hosted.weblate.org/engage/free-tube/) page to see what can be included. If enough translations have been added, let us know and we'll enable it to be used within FreeTube. Thank you very much to everyone who has translated for us in the past and we look forward to anyone's future contributions.

## New Contributors
* @Alhuin made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2859
* @miangraham made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2853
* @theforbiddenpool made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2893
* @gnuhead-chieb made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2895
* @elshimone made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2906
* @emmayjiang made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2964
* @ai25 made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2418
* @camcam1773 made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3126
* @lodwhet made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3116
* @kawanakaiku made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3008
* @ArturWagnerBusiness made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3172
* @predystopic-dev made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3203
* @arlm made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/2958
* @yarons made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3237
* @lamemakes made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3658
* @Rikthepixel made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3692
* @jadenet made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3710
* @trostboot made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3771
* @socherbyc made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3804
* @capecrusader-121 made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/3844

**Full Changelog**: https://github.com/FreeTubeApp/FreeTube/compare/v0.18.0-beta...v0.19.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.19.0-beta'>Release v0.19.0 Beta</a>.</em>

**Comment by S7venLights** [comment]
Wooo! so happy FreeTube is back! and working on wayland session xwayland :) Congrats

**Reply by S7venLights** [reply]
Oh I spoke too soon, the freetube app launches fine but video doesn't playback. Graphical glitches...

**Reply by absidue** [reply]
Try disabling variable refresh rate on your system, apparently electron doesn't play well with it on wayland: https://github.com/FreeTubeApp/FreeTube/issues/3664#issuecomment-1632233694

**Reply by absidue** [reply]
Alternatively if you have the setting enabled to proxy videos through Invidious, disable it as Invidious instances are currently getting IP blocked by YouTube but only for the video and audio streaming, everything else still works fine. (mentioned in the known issues in the release notes)

**Comment by ghost** [comment]
I am on a Raspberry Pi 4. RPI OS (32 bit). Freetube was always "slow" playing videos, you could see the frames etc. However, i am on the latest nightly right now and i have to say that there is a "huge improvement". Whatever you guys did just worked, please do not change it :) Videos are so smooth now.

PS: I am using "legacy formats" and 1080p. However, most videos play in 720p even when there is a 1080p video available in youtube. This is not new, it has always been the case. I just wanted to say. Can we do anything about this?

**Reply by absidue** [reply]
@sudo-splinter-cell The legacy formats are only available in 360p and 720p (720p legacy isn't available on all videos). If you want 1080p, you need to switch to the DASH formats, although they might run worse on a Raspberry Pi. The DASH formats are separate audio and video files, which get combined at play time, so they require more bandwidth and processing power than the legacy formats, which come with both audio and video in a single file.

**Comment by ghost** [comment]
After updating to the latest nightly, the smoothness is gone once again. I am starting to think that it was just my imagination that the videos were smoother. Maybe they never were.

**Reply by absidue** [reply]
@sudo-splinter-cell I'm sure they were but YouTube hates 3rd party clients and downloaders with a passion, so they keep making intentional changes to break stuff. There is already an open issue for it.

**Reply by ghost** [reply]
The videos are still smoothly playing in Chromium (which is specially tweaked for use with RPI). I wish there was a way to use Freetube inside a Chromium tab. I am pretty sure Freetube, as a seperate app, rocks on any other platform, but on a RPI, the juice is not enough.
